tests/cppException/enc/cppException.cpp

```diff
@@ -795,7 +802,53 @@ class BarClass08 : public std::exception
 
     ~BarClass08() throw()
     {
-        throw "X";
+        // throw "X";
+        // If the throw "X" is enabled above, then it triggers a code-generation
+        // bug with clang-8 and above. The bug happens only when Spectre
+        // mitigation (-llvm -x86-speculative-load-hardening) is enabled and
+        // optimizations are turned on. In such a case, the compiler inlines the
+        // destructor of BarClass08. The following code in ExceptionInUnwind
+        // function below
+        //      BarClass08 obj;
+        //      throw 'X';
+        // which is internally
+        //      BarClass08 obj;
+        //      throw 'X';
+        //      ~BarClass08();
+        // becoms
+        //      BarClass08 obj;
+        //      throw 'X';
+        //      throw "X";
+        // when ~BarClass08() is inlined.
+        // Speculative load hardening causes the following code to be emitted:
+        //      BarClass08 obj;
+        // (A)  throw 'X';
+        // (B)  assembly code to set rax to 0 in speculative execution path
+        //                                  1 in normal execution path
+        // (C)  or rax, rsp      # rax expected to be 0, rsp will be unchanged
+        //      throw "X";
+        // Due to what looks like incorrect exception unwinding information,
+        // when unwinding the callstack for exception thrown in (A), the control
+        // skips (B) and directly reaches (C).
+        // The value of RAX is non-zero since its setup code (B) has been
+        // skipped. (C) executes and corrupts RSP and subsequently when a callq
+        // instruction is executed for throw "X", a SIGBUS is raised.
+        //
+        // The expected behavior is that terminate is called due to uncaught
```

Comment:
I guess you mean
```suggestion
        // The expected behavior is that the destructor is called due to an uncaught
```
If I understand correctly, this sentence is trying to explain that what triggers the behavior explained in line 830-835. If so, consider rephrasing it to something like
```suggestion
       // This behavior is triggered when the destructor is called due to an uncaught
```

Reply:
The test-case locks down the scenario as stated in the comment. It expects that `terminate` is called due to an uncaught exception raised during unwinding of another exception.